src/components/axes/axis.ts

```diff
@@ -188,7 +188,7 @@ module Plottable {
     }
 
     protected _generateTickMarkAttrHash(isEndTickMark = false) {
-      var tickMarkAttrHash = {
+      var tickMarkAttrHash: { [key: string]: number | ((d: any) => number) } = {
```

Comment:
I could convert the hash values to a constant
```Typescript
() => 0
```
instead of just using `number`, but that complicates the later code, and d3 does accept the raw `number`s. I wanted to modify as little of the actual code logic as possible in this PR.

I should remove the `<any>`s on the following 4 lines though.